⚙️ 프로젝트 추가 설정

Legitgoons opened the pull request:
## 작업 이유
- 프로젝트 추가 설정
<br/>

## 작업 사항

### 1️⃣ eslint 추가 설정
- import order, fsd/import, @tanstack/query 3가지 패키지를 추가로 설치, 설정했습니다.
### 2️⃣ tanstack-query 설정
- QueryProvider.tsx 파일에 설정했습니다.
- react query devtools도 함께 설정했습니다.
### 3️⃣ tailwind css 설정
#### font 설정
- woff2 subset font를 기본 font로 설정하고, woff subset font를 fallback font로 사용했습니다.
- Figma에 정의된 대로 font를 선언해서 사용할 수 있도록 했습니다.
  - `text` + `Figma에 정의된 font name` + `font size의 첫 글자`를 입력해서 사용할 수 있습니다.
  ```tsx
  <div className="text-h1b">Hello World</div>;
  ```
![image](https://github.com/user-attachments/assets/1c85efd2-eabf-4b21-91f2-4c06fa5e9cf0)
```ts
const createFontSize = (
  size: string,
  fontWeight: string,
  lineHeight: string = '130%',
): [string, { letterSpacing: string; lineHeight: string; fontWeight: string | number }] => [
  size,
  { letterSpacing: '-0.03em', lineHeight, fontWeight },
];
//...
fontSize: {
  h1b: createFontSize('24px', 'bold'),
  h1s: createFontSize('24px', '600'),
  h2s: createFontSize('20px', '600'),
  h3b: createFontSize('16px', 'bold', '140%'),
  h3s: createFontSize('16px', '600', '140%'),
  h4s: createFontSize('14px', '600'),
  h4m: createFontSize('14px', '500'),
  b1s: createFontSize('12px', '600'),
  b1m: createFontSize('12px', '500'),
  b1r: createFontSize('12px', '400', '140%'),
  b2s: createFontSize('11px', '600'),
  b2m: createFontSize('11px', '500'),
  b2r: createFontSize('11px', '400'),
  b3m: createFontSize('9px', '500'),
  b4m: createFontSize('8px', '500'),
},
```
<br/>

## 리뷰어가 중점적으로 확인해야 하는 부분

- [ ] Eslint가 정상적으로 작동하는지
- [x] query devtools가 정상적으로 작동하는지
- [x] font가 정상적으로 적용되는지
  <br/>

## 발견한 이슈
- 현재 Local 개발 환경에서 eslint가 정상적으로 작동하지 않는 이슈가 있습니다. 혹시 다른 분들도 같은 이슈가 발생하신다면 review에 남겨주시면 감사드리겠습니다.
  - 병준님 덕분에 무사히 해결했습니다😉

Commit:
feat: zustand version 4.5.5 재설치

Commit:
feat: font 파일 추가

Commit:
feat: logo.svg 추가

Commit:
feat: network_error.png 추가

Commit:
feat: 기존 file 삭제

Commit:
feat: font 설정 완료

Commit:
feat: eslint 추가 설정

Legitgoons committed:
feat: tanstack/react-query devtool setting

Legitgoons committed:
feat: QueryProvider setting

Legitgoons committed:
feat: font-family update

Legitgoons committed:
fix: eslint unused setting remove

BangDori commented on src/app/provider/QueryProvider.tsx:
```diff
@@ -0,0 +1,15 @@
+'use client';
```
P4: QueryProvider에서 use client를 선언해주신 이유가 따로 있으신가요?

Legitgoons replied:
QueryClientProvider와 ReactQueryDevtools는 클라이언트 사이드에서 동작하기에, use client를 선언했습니다!

BangDori commented on tailwind.config.ts:
```diff
@@ -13,6 +13,26 @@ const config: Config = {
+        sans: ['Pretendard', 'sans-serif'],
+      },
+      fontSize: {
+        h1b: ['24px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: 'bold' }],
+        h1s: ['24px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '600' }],
+        h2s: ['20px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '600' }],
+        h3b: ['16px', { letterSpacing: '-0.03em', lineHeight: '140%', fontWeight: 'bold' }],
+        h3s: ['16px', { letterSpacing: '-0.03em', lineHeight: '140%', fontWeight: '600' }],
+        h4s: ['14px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '600' }],
+        h4m: ['14px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '500' }],
+        b1s: ['12px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '600' }],
+        b1m: ['12px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '500' }],
+        b1r: ['12px', { letterSpacing: '-0.03em', lineHeight: '140%', fontWeight: '400' }],
+        b2s: ['11px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '600' }],
+        b2m: ['11px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '500' }],
+        b2r: ['11px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '400' }],
+        b3m: ['9px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '500' }],
+        b4m: ['8px', { letterSpacing: '-0.03em', lineHeight: '130%', fontWeight: '500' }],
+      },
```
P4: 현재 `letterSpacing`값이 모두 `-0.03em`으로 동일한데 최상단에서 `letterSpacing` 값을 `-0.03em`으로 할당해준 후에 각 font 객체에서 `letterSpacing` 값을 축약하는 방법도 괜찮지 않을까요?

Legitgoons replied:
이거 저번에도 그랬는데 또 깜빡했군요,, 즉시 수정하겠습니다!

BangDori commented on .eslintrc.json:
P1: 로컬에서 확인해본 결과 eslint가 스크립트로 실행하면 동작하는데 IDE 상에서는 동작하고 있지 않았습니다. eslintrc.json과 eslint.config.mjs가 충돌해서 eslint.config.mjs만 적용되어 발생한 문제인 것 같습니다.

eslint 세팅 파일을 하나로 합치거나 eslint.config.mjs 세팅 파일을 제거해야 할 것 같습니다!

Legitgoons replied:
아하 한번 시도해보겠습니다 감사합니다!

BangDori reviewed (changes requested):
eslint 설정하신다구 고생하셨습니다! 리뷰 확인 부탁드려요~

suhwan2004 commented on tailwind.config.ts:
P4: 이거 코멘트 등록이 해당 줄에서 안되서 그러긴 한데,  저희 프로젝트내에서 mdx 사용이 없다면, content에 타겟을 ts, tsx로만 잡도록 하는건 어떨까요?

Legitgoons replied:
제가 알기로는 mdx가 storybook쪽에서 사용되기에 그대로 유지하는 것이 나을 것 같습니다. Js와 jsx는 삭제해도 큰 문제가 없을 듯 합니다.

다만 혹시 삭제하면 어떤 이점이 있을까요?

suhwan2004 replied:
어 혹시 mdx가 storybook에서 사용되는 케이스를 살짝 여쭤봐도 될까요? mdx 사용시에, 삭제가 필요없더라도
이점은 아닌데, 프로젝트 내에 사용이 되지 않는 파일까지 범위로 잡을 필요는 없을 것 같아서 코멘트 작성했던 것 같습니다.

Legitgoons replied:
제가 storybook 사용한 Repo를 다시 찾아봤는데, 사용하지 않았네요 ㅎㅎ;; 굳이 필요가 없다면 모두 삭제하겠습니다!

suhwan2004 reviewed (commented):
eslint, msw 이슈때문에 고생 많습니다... 든든합니다! 간단하게 리뷰 남겨두었어요

Legitgoons committed:
feat: tailwind css createFontSize 함수 구현

Legitgoons committed:
fix: eslint.config.mjs remove

eslint.config.mjs와 .eslintrc.json의 충돌로 인해 eslint가 정상적으로 작동하지 않아 삭제했습니다.

Legitgoons committed:
feat: eslint import update

React와 Next import문은 반드시 맨 위로 가도록 pathGroupsExcludedImportTypes에 추가했습니다.
사용하지 않는 styled component는 제거했습니다.

Legitgoons committed:
Merge branch 'main' into feat/project-setting-2

BangDori commented on tailwind.config.ts:
```diff
@@ -1,5 +1,14 @@
 import type { Config } from 'tailwindcss';
 
+const createFontSize = (
```
P5: createFontSize 메서드로 fontSize 값 설정하는거 너무 좋네요!

BangDori commented on .eslintrc.json:
```diff
@@ -1,8 +1,48 @@
+  "plugins": ["fsd-import", "@tanstack/query", "import"],
+  "extends": ["next/core-web-vitals", "next/typescript", "prettier", "plugin:storybook/recommended"],
+  "rules": {
+    "fsd-import/fsd-relative-path": "error",
+    "fsd-import/public-api-imports": "error",
+    "fsd-import/layer-imports": "error",
```
P4: 저희 프로젝트에서는 @ alias를 사용하고 있는데 fsd-import rules 설정에서 alias 옵션을 주지 않아도 정상적으로 작동하나요??

Legitgoons replied:
넵 따로 설정해주지않아도 문제없습니다

Legitgoons replied:
애초에 fsd import가 정상적으로 동작하지 않고 있습니다. 우선 해당 패키지를 삭제한 후 병합하겠습니다.
추후 msw설정을 완료한 후에, 다시 해결 방법을 찾아보겠습니다!

BangDori reviewed (approved):
고생하셨습니당~ eslint alias 설정만 확인 부탁드려요!

Legitgoons committed:
feat: tailwind.config.ts update

사용되지 않는 확장자 제거

Legitgoons committed:
feat: delete fsd-import
